### DISCUSSION TITLE
Feature request - Visually show Nodes to carry messages from its source to the next nodes

### DISCUSSION THREAD

**Discussion by ghost** [xyflow/xyflow#3266] [New Features]
**Current system** has edges that connect the nodes. This visually depicts that there is a flow of information from one node to another node.

**Proposal** when a node. receives a message from its source, visually show that the message is delivered to the next node or nodes with an animated object (circle or a dot) passing through the path of the edge.

In the current system, this is already possible with help of animated edges, though the speed can be controlled with the animation speed, having a object tracing the path to its next destination would conceptualised it and opens more options for creativity.

If not a feature request, how can I build something on top of react-flow's edge tracing object ?

**Comment by ghost** [comment]
For reference https://animejs.com/documentation/#motionPath

**Reply by bcakmakoglu** [reply]
It's possible using a cutom edge and `getPointAtLength`.

[Here's a very basic example of it.](https://codesandbox.io/s/immutable-star-5q62nh?file=/src/TransitionEdge.tsx)
